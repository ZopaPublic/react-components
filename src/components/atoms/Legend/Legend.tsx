import styled from 'styled-components';
import { spacing, typography } from '../../../constants';

const Legend = styled.legend`
  padding: 0;
  display: table;
  margin-bottom: ${spacing['2']};
  font-weight: ${typography.weights.semiBold};
  font-family: ${typography.primary};
  letter-spacing: 0;
`;

export default Legend;
