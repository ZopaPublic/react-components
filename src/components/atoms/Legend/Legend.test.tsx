import axe from '../../../../axe-helper';
import React from 'react';
import { render } from '@testing-library/react';
import Legend from './Legend';

describe('<Legend />', () => {
  it('renders the component with no a11y violations', async () => {
    const { container } = render(<Legend />);
    const results = await axe(container.innerHTML);

    expect(container.firstChild).toMatchSnapshot();
    expect(results).toHaveNoViolations();
  });
});
