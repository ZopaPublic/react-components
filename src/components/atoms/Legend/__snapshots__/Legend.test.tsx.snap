// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Legend /> renders the component with no a11y violations 1`] = `
.c0 {
  padding: 0;
  display: table;
  margin-bottom: 8px;
  font-weight: 600;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

<legend
  class="c0"
/>
`;
